candle-examples/examples/quantized-qwen2-instruct/main.rs

```diff
@@ -211,11 +223,18 @@ fn main() -> anyhow::Result<()> {
 
     let tokenizer = args.tokenizer()?;
     let mut tos = TokenOutputStream::new(tokenizer);
-    let prompt_str = args.prompt.unwrap_or_else(|| DEFAULT_PROMPT.to_string());
-    let prompt_str = format!(
-        "<|im_start|>user\n{}<|im_end|>\n<|im_start|>assistant\n",
-        prompt_str
-    );
+    let prompt_str = args
+        .prompt
+        .clone()
+        .unwrap_or_else(|| DEFAULT_PROMPT.to_string());
+    let prompt_str = if args.is_deepseek() {
```

Comment:
Below there is a pattern matching to check if the model is deepseek rather than a `if` test, we should probably decide between one of the two styles.

Reply:
I think pattern matching is more readable, but I'm fine with changing it if you'd prefer the `if` approach.